tests/integration/cmd_push_test.go

```diff
@@ -53,6 +53,94 @@ var _ = Describe("odo push command tests", func() {
 
 	})
 
+	Context("Check memory and cpu config before odo push", func() {
+		It("Should work when both minmemory and maxmemory is set..", func() {
+
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context, "--app", appName)
+
+			helper.CmdShouldPass("odo", "config", "set", "minMemory", "100Mi", "--context", context)
+			helper.CmdShouldPass("odo", "config", "set", "maxmemory", "200Mi", "--context", context)
+			helper.CmdShouldPass("odo", "push", "--context", context)
+		})
+
+		It("Should work when memory is set..", func() {
+
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context, "--app", appName)
+
+			helper.CmdShouldPass("odo", "config", "set", "Memory", "300Mi", "--context", context)
+			helper.CmdShouldPass("odo", "push", "--context", context)
+		})
+
+		It("Should fail if minMemory is set but maxmemory is not set..", func() {
+
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context, "--app", appName)
+
+			helper.CmdShouldPass("odo", "config", "set", "minmemory", "100Mi", "--context", context)
+			output := helper.CmdShouldFail("odo", "push", "--context", context)
+			Expect(output).To(ContainSubstring("`minmemory` should accompany `maxmemory` or use `odo config set memory` to use same value for both min and max or try not passing any of them"))
+		})
+
+		It("should fail if maxmemory is set but minmemory is not set..", func() {
+
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context, "--app", appName)
+
+			helper.CmdShouldPass("odo", "config", "set", "maxmemory", "400Mi", "--context", context)
+			output := helper.CmdShouldFail("odo", "push", "--context", context)
+			Expect(output).To(ContainSubstring("`minmemory` should accompany `maxmemory` or use `odo config set memory` to use same value for both min and max or try not passing any of them"))
+		})
+
+		It("Should work when both mincpu and maxcpu is set..", func() {
+
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context, "--app", appName)
+
+			helper.CmdShouldPass("odo", "config", "set", "mincpu", "0.2", "--context", context)
+			helper.CmdShouldPass("odo", "config", "set", "maxcpu", "0.5", "--context", context)
+			helper.CmdShouldPass("odo", "push", "--context", context)
+		})
+
+		It("Should work when cpu is set", func() {
```

Comment:
Cna it be done through code level validation i mean through UTs ?

Reply:
No. I don't think so, due to the same reason I mentioned above.